.dotnet/src/Custom/Files/IncrementalUploadJob.Protocol.cs

```diff
@@ -0,0 +1,208 @@
+using System.ClientModel;
+using System.ClientModel.Primitives;
+using System.Threading.Tasks;
+
+namespace OpenAI.Files;
+
+public partial class IncrementalUploadJob
+{
+    /// <summary>
+    /// <para>[Protocol Method]</para>
+    /// Adds a data part to the in-progress upload job. Data parts may be added in parallel and out of order, with
+    /// their sequencing determined by the final call to <see cref="Complete(string, BinaryContent, RequestOptions)"/>.
+    /// </summary>
+    /// <remarks>
+    /// </remarks>
+    /// <param name="jobId">
+    /// The ID of the upload job to add the data part to. This should typically be the value of <see cref="Id"/> in the
+    /// instance.
+    /// </param>
+    /// <param name="content">
+    /// <para>
+    /// The binary request content describing the data part. This is a multipart/form-data body that should use the same
+    /// boundary value described by <paramref name="contentType"/> and provide a required <c>data</c> form field.
+    /// </para>
+    /// An example value template:
+    /// <code>
+    /// BinaryContent.Create(BinaryData.FromString($$"""
+    ///     --{{boundaryValueMatchingContentType}}
+    ///     Content-Disposition: form-data; name="data"; filename="data_part"
+    ///
+    ///     {{contentForDataPart}}
+    ///     --{{boundaryValueMatchingContentType}}--
+    ///     """);
+    /// </code>
+    /// </param>
+    /// <param name="contentType">
+    /// <para>
+    /// The content type for the request. This should be multipart/form-data and provide a boundary value matched in
+    /// the payload of <paramref name="content"/>.
+    /// </para>
+    /// Example value:
+    /// <code>
+    /// $@"multipart/form-data; boundary=""{boundaryValueMatchingContent}"""
+    /// </code>
+    /// </param>
+    /// <param name="options">
+    /// Additional options for the protocol request, including cancellation and error behavior.
+    /// </param>
+    /// <returns>
+    /// A <see cref="ClientResult"/> with response content, retrievable via <see cref="ClientResult.GetRawResponse"/>,
+    /// that describes the newly added data part. The <c>id</c> from this part is necessary to sequence parts in the
+    /// final call to <see cref="Complete(string, BinaryContent, RequestOptions)"/>.
+    /// </returns>
+    public virtual Task<ClientResult> AddDataPartAsync(string jobId, BinaryContent content, string contentType, RequestOptions options)
```

Comment:
The `RequestOptions` parameter here and in every other other method that takes a `BinaryContent` parameter should be defaulted to `null`. Looking at this whole file, it seems like the only exceptions are `Cancel` and `CancelAsync`.

Reply:
thanks! fixed it (also the two in FileClient)